test/services/optout/OptOutServiceSpec.scala

```diff
@@ -109,45 +128,68 @@ class OptOutServiceSpec extends UnitSpec
       val data = UIJourneySessionData(
         sessionId = hc.sessionId.get.value,
         journeyType = OptOutJourney.Name,
-        optOutSessionData = Some(OptOutSessionData(None, selectedOptOutYear = Some(TaxYear.forYearEnd(forYearEnd).toString)))
+        optOutSessionData = Some(OptOutSessionData(None, selectedOptOutYear = None))
       )
       when(repository.get(any(), any())).thenReturn(Future.successful(Some(data)))
 
-      def reconfigureMock(): Future[Unit] = {
-        Future {
-          // Hmm... should we be mocking the repository?
-          Mockito.reset(repository)
-
-          // Are we just testing that mockito works here?
-          when(repository.set(any())).thenReturn(Future.successful(true))
-
-          val data = UIJourneySessionData(
-            sessionId = hc.sessionId.get.value,
-            journeyType = OptOutJourney.Name,
-            optOutSessionData = Some(OptOutSessionData(None, selectedOptOutYear = None))
-          )
-          when(repository.get(any(), any())).thenReturn(Future.successful(Some(data)))
-        }
-      }
-
       val f = for {
         isSaved <- service.saveIntent(TaxYear.forYearEnd(forYearEnd))
-        savedIntent <- service.fetchSavedIntent()
-        _ <- reconfigureMock()
-        // Does this test need more consistent data now we initialise the data with the context statuses?
-        isReset <- service.initialiseOptOutJourney(buildOneYearOptOutPropositionForPreviousYear(forYearEnd))
-        noneIntent <- service.fetchSavedIntent()
       } yield {
+        verify(repository).set(any())
         isSaved shouldBe true
-        savedIntent.isDefined shouldBe true
-        isReset shouldBe true
-        noneIntent.isDefined shouldBe false
       }
 
       f.futureValue shouldBe Succeeded
     }
   }
 
+  "OptOutService.recallOptOutInitialState" should {
+    "retrieve the intent year only" in {
```

Comment:
not clear what the intent of the test is and
can't see how the asserts in the test match the test name